### PR TITLE
Integration of type-1 MET corrections starting from raw MET

### PR DESCRIPTION
**Pull request by danbarto**
This PR makes several changes:
- jetmetUncertainties.py calculates type-1 MET corrections starting from rawMET, and applies JECs to AK4 jets. This module should work for both data (few more tests needed) and MC (2017 and 2018 tested).
- Corrections to larger cone size jets (e.g. AK8) that don't interfere with MET are separated off to jetUnceratinties.py

### CODE REVIEW

**Comment by camclean** [comment]
You should not separate the AK8 jets into a new file. Instead, you can just set corrMET to false for AK8 jets.

**Comment by danbarto** [comment]
While I do agree we shouldn't spread out into too many modules and try to keep unnecessary redundancy low, I'm not sure if we really want to have all the FatJet mass corrections and variations between the delicate type-1 MET correction code. It becomes messy, hard to read and difficult to debug. Anyway, I'll go through the code again and try to come up with a better solution.

Besides fixing the type-1 corrections (issue #122) that right now are just wrong, this PR also addresses issue #184. More input and comments are of course very welcome.

**Comment by ahinzmann** [comment]
For reference: This needs to be propagated to nanoaod-tools as well to be able to use JER V7:
https://github.com/cms-sw/cmssw/pull/28096/files

File: python/postprocessing/modules/jme/fatJetUncertainties.py [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+                    jet_msdcorr_nom           = jet_pt_jerNomVal*jet_msdcorr_jmrNomVal*jmsNomVal*jet_msdcorr_raw
+                    jets_msdcorr_nom    .append(jet_msdcorr_nom)
+                    jets_msdcorr_jerUp  .append(jet_pt_jerUpVal  *jet_msdcorr_jmrNomVal *jmsNomVal  *jet_msdcorr_raw)
+                    jets_msdcorr_jerDown.append(jet_pt_jerDownVal*jet_msdcorr_jmrNomVal *jmsNomVal  *jet_msdcorr_raw)
+                    jets_msdcorr_jmrUp  .append(jet_pt_jerNomVal *jet_msdcorr_jmrUpVal  *jmsNomVal  *jet_msdcorr_raw)
+                    jets_msdcorr_jmrDown.append(jet_pt_jerNomVal *jet_msdcorr_jmrDownVal*jmsNomVal  *jet_msdcorr_raw)
+                    jets_msdcorr_jmsUp  .append(jet_pt_jerNomVal *jet_msdcorr_jmrNomVal *jmsUpVal   *jet_msdcorr_raw)
+                    jets_msdcorr_jmsDown.append(jet_pt_jerNomVal *jet_msdcorr_jmrNomVal *jmsDownVal *jet_msdcorr_raw)
+
+                    #Also evaluated JMS&JMR SD corr in tau21DDT region: https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetWtagging#tau21DDT_0_43
```
Comment (fgolf) [inline python/postprocessing/modules/jme/fatJetUncertainties.py]: Should the values that follow be moved to jetmetHelper and set at creation?

File: python/postprocessing/modules/jme/jetmetUncertainties.py [hunk 2]
```diff
@@ -36,26 +35,19 @@ def __init__(self, era, globalTag, archive=None, globalTagProd=None,  jesUncerta
...
-        self.jmrVals = jmrVals
-        if not self.jmrVals:
-            print "WARNING: jmrVals is empty!!! Using default values. This module will soon be deprecated! Please use jetmetHelperRun2 in the future."
-            self.jmrVals = [1.0, 1.2, 0.8] #nominal, up, down
-            # Use 2017 values for 2018 until 2018 are released
-            if self.era in ["2017","2018"]:
-                self.jmrVals = [1.09, 1.14, 1.04] 
-
-        self.jetSmearer = jetSmearer(globalTag, jetType, self.jerInputFileName, self.jerUncertaintyInputFileName, self.jmrVals)
+        self.jetSmearer = jetSmearer(globalTag, jetType, self.jerInputFileName, self.jerUncertaintyInputFileName, [1.0, 1.2, 0.8])
```
[the remark this answers is not in the document]
Comment (danbarto) [reply python/postprocessing/modules/jme/jetmetUncertainties.py]: The values can in fact be removed, and we just use the default ones - JMR values shouldn't have an effect on the MET.

File: python/postprocessing/modules/jme/fatJetUncertainties.py [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+              
+              if self.doGroomed : 
+                  self.out.fillBranch("%s_msoftdrop_jes%sUp" % (self.jetBranchName, jesUncertainty), jets_msdcorr_jesUp[jesUncertainty])
+                  self.out.fillBranch("%s_msoftdrop_jes%sDown" % (self.jetBranchName, jesUncertainty), jets_msdcorr_jesDown[jesUncertainty])
+                
+            
+
+        return True
+
+# define modules using the syntax 'name = lambda : constructor' to avoid having them loaded when not needed
```
[the remark this answers is not in the document]
Comment (danbarto) [reply python/postprocessing/modules/jme/fatJetUncertainties.py]: yes, you are right

File: python/postprocessing/modules/jme/fatJetUncertainties.py (hunk 1 above)
Comment (danbarto) [reply python/postprocessing/modules/jme/fatJetUncertainties.py]: Can be done. What's your opinion, @camclean ?

**Comment by danbarto** [comment]
As future reference, validation and description of the algorithm: https://indico.cern.ch/event/854654/contributions/3594580/attachments/1924445/3184422/MET_type1corrections_nanoAODtools.pdf

**Pull request merged**